Sources/SlideButton/SlideButton.swift

```diff
@@ -6,33 +6,54 @@
     @Environment(\.isEnabled) private var isEnabled
     
-    private let title: String
+    private let title: Label
     private let callback: () async -> Void
     
+    public typealias Styling = SlideButtonStyling
     private let styling: Styling
     
     @GestureState private var offset: CGFloat
     @State private var swipeState: SwipeState = .start
     
-    /// Initializes a slide button with the given title, styling options, and callback.
+    /// Initializes a slide button with the given title, styling options, and action.
     ///
-    /// Use this initializer to create a new instance of `SlideButton` with the given title, styling, and callback. The `styling` parameter allows you to customize the appearance of the slide button, such as changing the size and color of the indicator, the alignment of the title text, and whether the text fades or hides behind the indicator. The `callback` parameter is executed when the user successfully swipes the indicator.
+    /// Use this initializer to create a new instance of `SlideButton` with the given title, styling, and callback. The `styling` parameter allows you to customize the appearance of the slide button, such as changing the size and color of the indicator, the alignment of the title text, and whether the text fades or hides behind the indicator. The `action` parameter is executed when the user successfully swipes the indicator.
     ///
     /// - Parameters:
-    ///   - title: The title of the slide button.
     ///   - styling: The styling options to customize the appearance of the slide button. Default is `.default`.
-    ///   - callback: The async callback that is executed when the user successfully swipes the indicator.
-    public init(_ title: String, styling: Styling = .default, callback: @escaping () async -> Void) {
-        self.title = title
-        self.callback = callback
+    ///   - action: The async callback action that is executed when the user successfully swipes the indicator.
+    ///   - label: The function creating a label view
+    public init(styling: Styling = .default, action: @escaping () async -> Void, @ViewBuilder label: () -> Label) {
+        self.title = label()
+        self.callback = action
         self.styling = styling
-        
+
         self._offset = .init(initialValue: styling.indicatorSpacing)
     }
     
+    @ViewBuilder
+    private var indicatorShape: some View {
+        switch styling.indicatorShape {
+        case .circular:
+            Circle()
+        case .rectangular(let cornerRadius):
+            RoundedRectangle(cornerRadius: max(0, (cornerRadius ?? 0) - styling.indicatorSpacing))
+        }
+    }
+    
+    @ViewBuilder
+    private var mask: some View {
+        switch styling.indicatorShape {
+        case .circular:
+            Capsule()
+        case .rectangular(let cornerRadius):
+            RoundedRectangle(cornerRadius: max(0, (cornerRadius ?? 0) - styling.indicatorSpacing))
```

Comment:
Only the indicatorShape needed the spacing subtracted. In the mask (which masks the background) should still be using `RoundedRectangle(cornerRadius: cornerRadius ?? 0)`

Reply:
ok, updated as well.